packages/sources/view-starknet-latest-answer/src/transport/function.ts

```
@@ -0,0 +1,117 @@
+    await sleep(context.adapterSettings.BACKGROUND_EXECUTE_MS)
+  }
+
+  async handleRequest(param: RequestParams) {
+    let response: AdapterResponse<StarknetLatestAnswerFunctionTransportTypes['Response']>
+    try {
+      response = await this._handleRequestStarknet(param)
+    } catch (e: unknown) {
+      const errorMessage = e instanceof Error ? e.message : 'Unknown error occurred'
+      logger.error(e, errorMessage)
+      response = {
+        statusCode: (e as AdapterInputError)?.statusCode || 502,
+        errorMessage,
+        timestamps: {
+          providerDataRequestedUnixMs: 0,
+          providerDataReceivedUnixMs: 0,
+          providerIndicatedTimeUnixMs: undefined,
+        },
+      }
+    }
+    await this.responseCache.write(this.name, [{ params: param, response }])
+  }
+
+  async _handleRequestStarknet(
+    param: RequestParams,
+  ): Promise<AdapterResponse<StarknetLatestAnswerFunctionTransportTypes['Response']>> {
+    const { address } = param
+    const { STARKNET_RPC_URL } = config.settings
+
+    const rpcUrl = STARKNET_RPC_URL
+
+    if (!rpcUrl) {
+      throw new AdapterInputError({
+        statusCode: 400,
+        message: `Missing RPC URL: '${rpcUrl}' environment variables.`,
+      })
+    }
+
+    if (!this.provider) {
+      this.provider = new RpcProvider({ nodeUrl: rpcUrl })
+    }
+
+    const callData = {
+      contractAddress: address,
+      entrypoint: 'latest_round_data',
+      calldata: [],
+    }
+
+    const providerDataRequestedUnixMs = Date.now()
+
+    let res
+    try {
+      // res: [round_id, answer, block_num, started_at, updated_at]
+      res = await this.provider.callContract(callData)
+    } catch (e) {
+      throw new AdapterInputError({
+        statusCode: 502,
+        message: `RpcProvider.callContract Failed - ${e}`,
+      })
+    }
+    // extract field "answer"
+    const answer = num.hexToDecimalString(res[1])
```

Can we throw an error if this is undefined or not a decimal?